### PR TITLE
Check uniqueness of clientId when restoring an application 

### PR DESCRIPTION
**Pull request by gaetanmaisse**
**Issue**

https://github.com/gravitee-io/issues/issues/6380

**Description**

 - Use constants instead of duplicating hardcoded strings
 - Check uniqueness of clientId when restoring an application 

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.blob.core.windows.net/6380-fix-app-restore/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

### CODE REVIEW

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java [hunk 1]
```diff
@@ -717,6 +707,10 @@ public ApplicationEntity restore(String applicationId) {
                 throw new ApplicationActiveException(application.getName());
             }
 
+            if (application.getMetadata() != null && application.getMetadata().containsKey("client_id")) {
+                checkClientIdIsUniqueForEnv(application.getEnvironmentId(), application.getMetadata().get("client_id"));
+            }
```
Comment (Okhelifi) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java]: Can we use the constant Application.METADATA_CLIENT_ID here ?

**Review by Okhelifi — APPROVED** [review]
👍🏼 lgtm

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java (hunk 1 above)
Comment (gaetanmaisse) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java]: Good catch!

Comment (marcambier) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java]: done :heavy_check_mark:

**Review by marcambier — APPROVED** [review]

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java (hunk 1 above)
Comment (gaetanmaisse) [reply gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApplicationServiceImpl.java]: Thanks a lot @marcambier !

**Pull request merged**